asio/cmake/variables.cmake

```diff
@@ -0,0 +1,24 @@
+# ---- Developer mode ----
+
+# Developer mode enables targets and code paths in the CMake scripts that are
+# only relevant for the developer(s) of asio
+# Targets necessary to build the project must be provided unconditionally, so
+# consumers can trivially build and package the project
+if(PROJECT_IS_TOP_LEVEL)
+  option(asio_DEVELOPER_MODE "Enable developer mode" ON)
```

Comment:
This must be `OFF` by default. Users must not be bothered by this project's developer mode.